### ISSUE TITLE
Dynamic Color Picker with ACF not working in Posts element

### ISSUE THREAD

**Issue by wagnerch1981** [elementor/elementor#10991]
## Description

In the "Posts"-Element the color-picker is not working with "ACF Color Picker Field".
I've tried so far the following color picker options: card > background color, card > border color, content > title
Neither in the frontend nor in the backend the color shows up.

## Steps to reproduce

ACF 
> setup ACF Color-Picker Field for Posts

POSTS 
> select the color in each post with the color-picker field

ELEMENTOR 
> drag & drop "posts" element in page
> setup query
> select in style > "background color" the ACF-Color-Picker field

## Isolating the problem

- [x] This bug happens with only Elementor, Elementor Pro, ACF, ACF Pro plugin active
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
					
== Server Environment ==
	Operating System: Linux
	Software: Apache/2.4.10
	MySQL version: MySQL Community Server (GPL) v5.7.27
	PHP Version: 7.2.26-1+0~20191218.33+debian8~1.gbpb5a34b
	PHP Max Input Vars: 1000
	PHP Max Post Size: 20M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.3.2
	Site URL: https://partner.hausarzt.digital
	Home URL: https://partner.hausarzt.digital
	WP Multisite: No
	Max Upload Size: 20 MB
	Memory limit: 256M
	Permalink Structure: /%category%/%postname%/
	Language: de-DE
	Timezone: Europe/Berlin
	Debug Mode: Inactive

== Theme ==
	Name: Content Hub
	Version: 1.0
	Author: Christian Wagner
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.2.2
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: de_DE
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36

== Active Plugins ==
	ACF Content Analysis for Yoast SEO
		Version: 2.3.0
		Author: Thomas Kräftner, ViktorFroberg, marol87, pekz0r, angrycreative, Team Yoast

	Advanced Custom Fields
		Version: 5.8.9
		Author: Elliot Condon

	Advanced Custom Fields PRO
		Version: 5.8.8
		Author: Elliot Condon

	All-in-One WP Migration
		Version: 7.18
		Author: ServMask

	Better Search Replace
		Version: 1.3.3
		Author: Delicious Brains

	Custom Post Type Permalinks
		Version: 3.3.4
		Author: Toro_Unit

	Custom Post Type UI
		Version: 1.7.4
		Author: WebDevStudios

	Duplicate Post
		Version: 3.2.4
		Author: Enrico Battocchi

	Dynamic Content for Elementor
		Version: 1.8.7.2
		Author: Dynamic.ooo

	Elementor
		Version: 2.9.7
		Author: Elementor.com

	Elementor Pro
		Version: 2.9.2
		Author: Elementor.com

	Map user caps for SSO by Weslink
		Version: 1.0.0
		Author: Weslink GmbH

	Members
		Version: 2.2.0
		Author: MemberPress

	Simple Single Sign On
		Version: 1.4.6
		Author: Dash10 Digital

	Wordfence Security
		Version: 7.4.6
		Author: Wordfence

	WP Fastest Cache
		Version: 0.9.0.4
		Author: Emre Vona

	Yoast SEO
		Version: 13.3
		Author: Team Yoast


== Elements Usage ==
	
	footer : 1
		divider : 1
		heading : 5
		nav-menu : 5
		text-editor : 1
	header : 2
		heading : 2
		image : 2
		nav-menu : 1
	single : 6
		global : 1
		heading : 6
		image : 3
		posts : 1
		text-editor : 7
		theme-post-content : 1
		video : 1
	widget : 0
		posts : 31
	wp-page : 1
		heading : 2
		posts : 1
		text-editor : 1


== Log ==
	
JS: showing 6 of 6JS: 2020-03-23 14:57:46 [error X 3][https://partner.hausarzt.digital/wp-content/plugins/elementor-pro/assets/js/editor.min.js?ver=2.8.3:2:54590] Cannot read property \'top\' of undefined 
JS: 2020-03-23 17:35:15 [error X 1][https://partner.hausarzt.digital/wp-content/plugins/elementor/assets/js/editor-document.min.js?ver=2.9.6:2:187454] Cannot read property \'map\' of undefined 
JS: 2020-03-26 12:52:52 [error X 31][https://partner.hausarzt.digital/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.9.7:2:73316] Cannot read property \'select2\' of undefined 
JS: 2020-03-26 12:54:43 [error X 1][https://partner.hausarzt.digital/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view8788\") has already been destroyed and cannot be used. 
JS: 2020-03-27 08:06:48 [error X 2][https://partner.hausarzt.digital/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=2.9.7:2:58852] Cannot read property \'attributes\' of undefined 
JS: 2020-03-27 09:45:40 [error X 1][https://partner.hausarzt.digital/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view198\") has already been destroyed and cannot be used. 

Log: showing 4 of 42020-03-23 16:12:39 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.8.3',
  'to' => '2.9.1',
)]
2020-03-26 13:35:26 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.9.6',
  'to' => '2.9.7',
)]
2020-03-26 13:35:30 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.9.6',
  'to' => '2.9.7',
)]
2020-03-26 13:36:17 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.9.1',
  'to' => '2.9.2',
)]

PHP: showing 3 of 3PHP: 2020-03-23 17:55:41 [notice X 39][/www/partner.hausarzt.digital/documents/wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-text.php::34] Undefined offset: 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-03-23 18:20:19 [error X 2][/www/partner.hausarzt.digital/documents/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::259] Uncaught Error: Call to a member function update_meta() on boolean in /www/partner.hausarzt.digital/documents/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:259
Stack trace:
#0 /www/partner.hausarzt.digital/documents/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(180): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(NULL, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /www/partner.hausarzt.digital/documents/wp-content/plugins/elementor/core/common/modules/ajax/module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 /www/partner.hausarzt.digital/documents/wp-includes/class-wp-hook.php(288): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 /www/partner.hausarzt.digital/documents/wp-includes/class-wp-h [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-03-27 11:44:27 [notice X 2][/www/partner.hausarzt.digital/documents/wp-content/plugins/elementor-pro/modules/query-control/module.php::736] Trying to get property 'name' of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]


				
</details>

**Comment by lior-sh** [comment]
Hi @wagnerch1981,

Thanks for your feedback
After some investigation, this issue happens also with the native ACF shortcode.
When accessing the preview mode the post_id changes due to \acf_revisions::acf_validate_post_id action hook and therefore this is not a bug related to Elementor but how ACF outputs its fields on the WP preview.
Thanks!

**Comment by wagnerch1981** [comment]
@lior-sh 

Thank you for investigating!

You write that the problem is caused by how ACF outputs its fields on the WP-Preview. But the problem also occurs in the regular (non-preview) output. The dynamic color never shows up. Is there any way I can get this working?

**Comment by Marceloolz** [comment]
Hello! My problem was putting a dynamic color.

I managed to solve it without extra plugins, just doing it as follows:

In attributes, I placed the acf color field, and in its settings, in

after I filled it in like this:
style|background-color:

Before: !important

I hope it works for you too! Good luck.

![image](https://github.com/elementor/elementor/assets/19634278/7e9e23b1-e203-4a4a-905c-80fa1e67d28c)
